examples/azure/terraform-azure-keyvault-example/main.tf

```diff
@@ -0,0 +1,157 @@
+# DEPLOY A KEY VAULT
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_key_vault" "key_vault" {
+  name                        = "keyvault-${var.postfix}"
+  location                    = azurerm_resource_group.resource_group.location
+  resource_group_name         = azurerm_resource_group.resource_group.name
+  enabled_for_disk_encryption = true
+  tenant_id                   = data.azurerm_client_config.current.tenant_id
+  soft_delete_enabled         = true
+  soft_delete_retention_days  = 7
+  purge_protection_enabled    = false
+
+  sku_name = "standard"
+
+  access_policy {
+    tenant_id = data.azurerm_client_config.current.tenant_id
+    object_id = data.azurerm_client_config.current.object_id
+
+    key_permissions = [
+      "create",
+      "get",
+      "list",
+      "delete",
+    ]
+
+    secret_permissions = [
+      "set",
+      "get",
+      "list",
+      "delete",
+    ]
+
+    certificate_permissions = [
+      "create",
+      "delete",
+      "deleteissuers",
+      "get",
+      "getissuers",
+      "import",
+      "list",
+      "listissuers",
+      "managecontacts",
+      "manageissuers",
+      "setissuers",
+      "update",
+    ]
+  }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A SECRET TO THE KEY VAULT
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_key_vault_secret" "key_vault_secret" {
+  name         = "${var.secret_name}-${var.postfix}"
+  value        = "mysecret"
+  key_vault_id = azurerm_key_vault.key_vault.id
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+#  DEPLOY A KEY TO THE KEY VAULT
+# ---------------------------------------------------------------------------------------------------------------------
+
+resource "azurerm_key_vault_key" "key_vault_key" {
+  name         = "${var.key_name}-${var.postfix}"
+  key_vault_id = azurerm_key_vault.key_vault.id
+  key_type     = "RSA"
+  key_size     = 2048
+
+  key_opts = [
+    "decrypt",
+    "encrypt",
+    "sign",
+    "unwrapKey",
+    "verify",
+    "wrapKey",
+  ]
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+#  DEPLOY A CERTIFICATE TO THE KEY VAULT
+# ---------------------------------------------------------------------------------------------------------------------
+resource "azurerm_key_vault_certificate" "key_vault_certificate" {
+  name         = "${var.certificate_name}-${var.postfix}"
+  key_vault_id = azurerm_key_vault.key_vault.id
+
+  certificate {
+    contents = filebase64("example.pfx")
+    password = "password"
+  }
```

Comment:
Add a comment indicating that we have the pfx file and password in plain text for this example to make it easier to test, but in production modules one should use a more secure mechanisms for transferring these files.

Reply:
done